lib/plugins/inspec-streaming-reporter-progress-bar/lib/inspec-streaming-reporter-progress-bar/streaming_reporter.rb

```diff
@@ -28,85 +32,128 @@ class StreamingReporter < Inspec.plugin(2, :streaming_reporter)
+      title = notification.example.metadata[:title]
+      full_description = notification.example.metadata[:full_description]
+      control_impact = notification.example.metadata[:impact]
+      set_status_mapping(control_id, "error")
+      show_progress(control_id, title, full_description, control_impact) if control_ended?(control_id)
+    end
+
+    def example_na(notification)
+      control_id = notification.example.metadata[:id]
+      title = notification.example.metadata[:title]
+      full_description = notification.example.metadata[:full_description]
+      control_impact = notification.example.metadata[:impact]
+      set_status_mapping(control_id, "na")
+      show_progress(control_id, title, full_description, control_impact) if control_ended?(control_id)
     end
 
     private
 
-    def show_progress(control_id)
+    def show_progress(control_id, title, full_description, control_impact)
       @bar ||= ProgressBar.new(controls_count, :bar, :counter, :percentage)
       sleep 0.1
       @bar.increment!
-      @bar.puts format_it(control_id)
-    rescue Exception => ex
-      raise "Exception in Progress Bar streaming reporter: #{ex}"
+      @bar.puts format_it(control_id, title, full_description, control_impact)
+    rescue StandardError => e
+      raise "Exception in Progress Bar streaming reporter: #{e}"
     end
 
-    def format_it(control_id)
-      control_status = if @status_mapping[control_id].include? "failed"
+    def format_it(control_id, title, full_description, control_impact)
+      control_status = if full_description.match?(/Source Code Error/)
+                         "error"
+                       elsif control_impact == 0
+                         "na"
+                       elsif @status_mapping[control_id].uniq.include?("failed")
                          "failed"
-                       elsif @status_mapping[control_id].include? "skipped"
+                       elsif @status_mapping[control_id].uniq.include?("skipped") && @status_mapping[control_id].uniq.one?
                          "skipped"
-                       elsif @status_mapping[control_id].include? "passed"
+                       elsif @status_mapping[control_id].uniq.any? { |val| /"passed"||"skipped"/ =~ val }
                          "passed"
                        end
-
       indicator = INDICATORS[control_status]
       message_to_format = ""
       message_to_format += "#{indicator}  "
-      message_to_format += control_id.to_s.lstrip.force_encoding(Encoding::UTF_8)
+      message_to_format += "#{control_id.to_s.lstrip.dup.force_encoding(Encoding::UTF_8)}  "
+      message_to_format += "#{title.gsub(/\n*\s+/, " ").to_s.force_encoding(Encoding::UTF_8)}  " if title
+      message_to_format += "#{full_description.gsub(/\n*\s+/, " ").to_s.force_encoding(Encoding::UTF_8)}  " unless title
       format_with_color(control_status, message_to_format)
+    rescue StandardError => e
+      raise "Exception in show_progress: #{e}"
```

Comment:
I think we should have error messages that could be also understood by an inspec user. 
So putting `show_progress` and `format_with_color` does help a developer to debug. But maybe not so useful for the user. Maybe use a message that is useful to both user and a developer? Makes sense?

Reply:
Happy to, what kind of message should we provide do you think? Parse `e` to bubble up the issue or put a case statment in that captures the N common errors that can happen at this stage and 'decode' them into nature language for the end user.

Reply:
> I think we should have error messages that could be also understood by an inspec user.

When it comes to InSpec profiles, we use the following short phrases to qualify what each outcome means to the user, so they are not confused as to what has happened and "whose fault is it?":

Passed: "All sub-tests for this control have passed or were skipped"
Failed: "At least one sub-test failed for this control"
Not Applicable: "System exception or absent component"
Not Reviewed: "Can only be tested manually at this time"
Profile Error: "Errors running test - check profile run privileges or check with the author of profile."

Could these somehow be added to the enhanced reporting engine?  Thanks!

Reply:
@aaronlippold Before you made this change what I had here was `Exception in Progress Bar streaming reporter`. Nothing fancy but somehow clear to the user where it went wrong? So maybe mix of this and the new change of message that you have made. Sounds good? 

And for examples, you can search with `exception` keep word in this repo and you will get an idea how we have framed messages in case of exceptions.